client/request.go

```diff
@@ -110,13 +110,20 @@ func (cli *Client) sendClientRequest(ctx context.Context, method, path string, q
 		if !cli.transport.Secure() && strings.Contains(err.Error(), "malformed HTTP response") {
 			return serverResp, fmt.Errorf("%v.\n* Are you trying to connect to a TLS-enabled daemon without TLS?", err)
 		}
+
 		if cli.transport.Secure() && strings.Contains(err.Error(), "remote error: bad certificate") {
 			return serverResp, fmt.Errorf("The server probably has client authentication (--tlsverify) enabled. Please check your TLS client certification settings: %v", err)
 		}
 
 		return serverResp, fmt.Errorf("An error occurred trying to connect: %v", err)
 	}
 
+	if resp != nil {
+		serverResp.statusCode = resp.StatusCode
+	}
+
+	defer resp.Body.Close()
```

Comment:
I don't think we should close the body here. Functions that stream responses depend on it to not be closed.